internal/cmd/auth/update-kubeconfig.go

```diff
@@ -0,0 +1,94 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+func updateKubeconfigCmd() *cobra.Command {
+	var kubeconfig, baseURL, projectName string
+
+	cmd := &cobra.Command{
+		Use:   "update-kubeconfig",
+		Short: "Update the kubeconfig file",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Determine kubeconfig path
+			var kubeconfigPath string
+			if kubeconfig != "" {
+				kubeconfigPath = kubeconfig
+			} else if envKC := os.Getenv("KUBECONFIG"); envKC != "" {
+				kubeconfigPath = envKC
+			} else {
+				kubeconfigPath = clientcmd.RecommendedHomeFile
+			}
+
+			serverURL, err := url.Parse(baseURL)
+			if err != nil {
+				return fmt.Errorf("failed to parse base URL option: %w", err)
+			}
+
+			if projectName != "" {
+				serverURL.Path = "/control-plane/v1alpha/projects/" + projectName
```

Comment:
Would this be better as the following path?

```suggestion
				serverURL.Path = "/v1alpha/projects/" + projectName + "/control-plane"
```

Reply:
I figured we'd want the URL to be namespaced and avoid a top level version parameter, but I'm open to your suggestion here.

Reply:
Ahh, right.. How about we follow the path structure that will exist for projects in the datum core api, and expect that an ingress layer forwards the `/control-plane` path to the correct backend?

```suggestion
				serverURL.Path = "/apis/resourcemanager.datumapis.com/v1alpha/projects/"+projectName+"/control-plane"
```

Reply:
I like it. Updated the code to use this path.